src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolTargetingNonSupportedTFM.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.Build.Tasks;
+using System;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToPackAToolTargetingNonSupportedTFM : SdkTest
+    {
+
+        public GivenThatWeWantToPackAToolTargetingNonSupportedTFM(ITestOutputHelper log) : base(log)
+        {
+
+        }
+
+        [Theory]
+        // lower than netcoreapp2.0
+        [InlineData("TargetFramework", "netcoreapp2.0", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        [InlineData("TargetFramework", "netcoreapp1.1", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        [InlineData("TargetFrameworks", "netcoreapp2.0;netcoreapp2.1", "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")]
+        // non netcoreapp
+        [InlineData("TargetFramework", "netstandard2.0", "PackAsToolOnlySupportNetcoreapp")]
+        public void It_should_fail_with_error_message(string targetFrameworkProperty, string targetFramework, string error)
+        {
+            TestAsset helloWorldAsset = _testAssetsManager
+                                        .CopyTestAsset("PortableTool", "PackNonSupportedTFM")
+                                        .WithSource()
+                                        .WithProjectChanges(project =>
+                                        {
+                                            XNamespace ns = project.Root.Name.Namespace;
+                                            XElement propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+
+                                            propertyGroup.Element(ns + "TargetFramework").Remove();
+                                            propertyGroup.Add(new XElement(ns + targetFrameworkProperty, targetFramework));
+
+                                        })
+                                        .Restore(Log);
+            var packCommand = new PackCommand(Log, helloWorldAsset.TestRoot);
+
+            CommandResult result = packCommand.Execute();
+            result.ExitCode.Should().NotBe(0);
+
+            // walk around attribute requires static
+            string message = null;
+            if (error == "PackAsToolDoesNotSupportTFMLowerThanNetcoreapp21")
```

Comment:
I cannot find a better way to do it

Reply:
Can you use the same code that is generated for you in the properties?  IE:

```C#
string message = Microsoft.NET.Build.Tasks.Strings.ResourceManager.GetString(error);
```

Also I'd rename `error` to something like `expectedErrorResourceName` and `message` to `expectedErrorMessage`.

Reply:
Changed as suggested